src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/ValidationRules/RequiresPermissionValidationRule.cs

```diff
@@ -16,114 +13,69 @@ public class RequiresPermissionValidationRule : IValidationRule
     {
         public static readonly string ErrorCode = "Unauthorized";
         private readonly IAuthorizationService _authorizationService;
-        private readonly IStringLocalizer<RequiresPermissionValidationRule> S;
+        private readonly IStringLocalizer<RequiresPermissionValidationRule> _localizer;
 
-        public RequiresPermissionValidationRule(IAuthorizationService authorizationService, IStringLocalizer<RequiresPermissionValidationRule> s)
+        public RequiresPermissionValidationRule(IAuthorizationService authorizationService, IStringLocalizer<RequiresPermissionValidationRule> localizer)
         {
             _authorizationService = authorizationService;
-            S = s;
+            _localizer = localizer;
         }
 
-        public async Task<INodeVisitor> ValidateAsync(ValidationContext validationContext)
+        public Task<INodeVisitor> ValidateAsync(ValidationContext validationContext)
         {
-            var operationType = OperationType.Query;
-            var userContext = (GraphQLUserContext)validationContext.UserContext;
+            var graphQLContext = (GraphQLUserContext)validationContext.UserContext;
 
-            return await Task.FromResult(new NodeVisitors(
-                new MatchingNodeVisitor<Operation>(async (astType, validationContext) =>
+            return Task.FromResult((INodeVisitor)new NodeVisitors(
+                new MatchingNodeVisitor<Operation>((astType, context) =>
                 {
-                    operationType = astType.OperationType;
-                    await AuthorizeOperationAsync(astType, validationContext, userContext, operationType, astType.Name);
+                    if (astType.OperationType == OperationType.Mutation)
+                    {
+                        if (!_authorizationService.AuthorizeAsync(graphQLContext.User, Permissions.ExecuteGraphQLMutations).GetAwaiter().GetResult())
```

Comment:
the perf gains in the other version are that it actually uses the async api ... this is calling sync